### PR TITLE
fixing exec call to be more robust

### PR DESCRIPTION
**Pull request by asmodehn**
regarding http://bugs.python.org/issue16781

Regarding problems such as : http://answers.ros.org/question/221308/third-party-release-of-python-package/

### CODE REVIEW

File: cmake/interrogate_setup_dot_py.py [hunk 1]
```diff
@@ -241,7 +241,8 @@ def main():
             pass
 
         with open(args.setupfile_path, 'r') as fh:
-            exec(fh.read())
+            exec(fh.read(),{})
```
Comment (dirk-thomas) [inline cmake/interrogate_setup_dot_py.py]: Please add a space after the comma (PEP8).

File: cmake/interrogate_setup_dot_py.py [hunk 2]
```diff
@@ -241,7 +241,8 @@ def main():
...
+
```
Comment (dirk-thomas) [inline cmake/interrogate_setup_dot_py.py]: Please remove the added empty line.

**Comment by dirk-thomas** [comment]
Please comment on the PR after making changes. Otherwise there is no notification email to the maintainers.

**Comment by dirk-thomas** [comment]
Please comment on the PR after making changes. Otherwise there is no notification email to the maintainers.

Can you please squash the commits.

**Comment by asmodehn** [comment]
Done. see https://github.com/ros/catkin/pull/766. Closing this one.

**Comment by dirk-thomas** [comment]
In the future you can simply squash the commits on the existing branch and force push them and the PR will be updated automatically.

**Comment by asmodehn** [comment]
That works only if noone else has cloned my branch ( which I cannot actually know ).
Otherwise rewriting history will break on their side next time they ll try to pull.

**Comment by wjwwood** [comment]
@asmodehn That's true, but no one should be using this pull request branch directly so it doesn't matter. You only need to worry about that on target branches (e.g. `master`, `indigo-devel`, etc.).

**Pull request closed without merging**